seed/seedwriter/writer.go

```diff
@@ -373,32 +480,42 @@ func (w *Writer) SnapsToDownload() (snaps []*SeedSnap, err error) {
 	}
 
 	snapsFromModel := make([]*SeedSnap, 0, len(modSnaps))
+	toDownload := make([]*SeedSnap, 0, len(modSnaps))
 
-	// XXX local snaps, extra snaps
+	// XXX extra snaps
 
 	for _, modSnap := range modSnaps {
-		optSnap, _ := w.byNameOptSnaps.Lookup(modSnap).(*OptionSnap)
+		sn, _ := w.byRefLocalSnaps.Lookup(modSnap).(*SeedSnap)
+		var optSnap *OptionSnap
+		if sn == nil {
+			// not local, to download
+			optSnap, _ = w.byNameOptSnaps.Lookup(modSnap).(*OptionSnap)
+			sn = &SeedSnap{
+				SnapRef: modSnap,
+
```

Comment:
I know it existed already in the code, but perhaps newline is not neccessary here.

Reply:
it's spllit the exported vs not exported fields, what I did is doing the same in the other construction site